crds/workspace.devfile.io_devworkspaces.yaml

```diff
@@ -4605,6 +4612,118 @@ spec:
+                            for outer loop builds
+                          oneOf:
+                          - required:
+                            - dockerfile
+                          properties:
+                            dockerfile:
+                              description: Allows specifying dockerfile type build
+                              oneOf:
+                              - required:
+                                - uri
+                              - required:
+                                - registry
+                              - required:
+                                - git
+                              properties:
+                                args:
+                                  description: The arguments to supply to the dockerfile
+                                    build.
+                                  items:
+                                    type: string
+                                  type: array
+                                buildContext:
+                                  description: Path of source directory to establish
+                                    build context. Defaults to ${PROJECT_ROOT} in
+                                    the container
+                                  type: string
+                                git:
+                                  description: Dockerfile's Git source
+                                  properties:
+                                    checkoutFrom:
+                                      description: Defines from what the project should
+                                        be checked out. Required if there are more
+                                        than one remote configured
+                                      properties:
+                                        remote:
+                                          description: The remote name should be used
+                                            as init. Required if there are more than
+                                            one remote configured
+                                          type: string
+                                        revision:
+                                          description: The revision to checkout from.
+                                            Should be branch name, tag or commit id.
+                                            Default branch is used if missing or specified
+                                            revision is not found.
+                                          type: string
+                                      type: object
+                                    gitLocation:
+                                      description: Location of the Dockerfile in the
+                                        Git repository when using git as Dockerfile
+                                        src.
+                                      type: string
+                                    remotes:
```

Comment:
Does dockerfile from git source really have a case for multiple origins?

Reply:
I shall update this to be like starterProject, ie it can have at most only 1 remote and put some validation for it

Reply:
updated with validation!